Learning | 2020-07-07 (#316)
emberjs/core-notes: learning-team/2020-07/2020-07-07.md

# July 7, 2020

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [x] Chris Manson (CM)
- [x] Isaac Lee (IL)
- [x] Jared Galanis (JG)
- [x] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [x] Ricardo Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests
- [x] James Davis

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

> - locks has merged a script for doing guides releases!
> - locks has a first version of a bot for the releases page

Reviewer: ```suggestion
- RM has merged a script for doing guides releases!
- RM has a first version of a bot for the releases page
```

- CM and JG have continued working on the discord bot to remind people of releases. Will share updates!

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

- CM - in the process of integrating the Discourse Comments into the blog. But it doesn't fit due to theming of the forum. So we added a new card to the website redesign to update the Discourse forum theme.

### Review PR queue

See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

> ### release scripts and automation (RM, 15mins)

Reviewer: ```suggestion
### Release scripts and automation (RM, 15mins)
```

RM: There are 3 things to address here.
- automating releases for each repository. We have done some of that.
- making sure that we know how to release the thing if the automation fails or we have a special case
- a tool to do the entire release

> Want to bring this up to the group because of something CM brough up. The tool is written in Rust because Node scripts are a pain in comparison. On the Guides source repo, the automation is in bash since it's the most portable. Some people may not be familiar but it's the lingua franca and requires no extra installs. API docs are Node scripts. I'm bringing this up so that we can discuss what should be written and how.

Reviewer: ```suggestion
Want to bring this up to the group because of something CM brought up. The tool is written in Rust because Node scripts are a pain in comparison. On the Guides source repo, the automation is in bash since it's the most portable. Some people may not be familiar but it's [bash is] the lingua franca and requires no extra installs. API docs are Node scripts. I'm bringing this up so that we can discuss what should be written and how.
```

> CM: For truly internal tools, we have more flexibility. For things we want to be able to allow people to contribute to, we should use JavaScript. There are tools like shell.js that can help here. The current situation for the guides being difficult to release are due to unfinished work. There are project cards for it.

Reviewer: ```suggestion
CM: For truly internal tools, we have more flexibility. For things we want to be able to allow people to contribute to, we should use JavaScript. There are tools like shell.js that can help here. The current situation for the guides being difficult to release is due to unfinished work. There are project cards for it.
```

RM: They are not hard to release, just they have some manual steps.

> CM: ideally it should be Ember deploy Algolia

Reviewer: ```suggestion
CM: ideally it should be `ember deploy algolia`
```

RM: We have to clear the Algolia index for each commit in master, right?

CM/JW: no

RM: Otherwise they would be out of date?

CM: yes, and no. We generally do this only on releases. Otherwise we would run out of indexes. Do you mean the latest release or all of them?

RM: The latest, because it keeps changing

> JW: I would not say no to donated work done in Rust. OSS is an opportunity for contributors to learn and grow, and that includes us. I would like if there was 1 other person who is active, who we can ask questions of. And also we could schedule a code walkthrough so that we can see how easy it would be for someone who doesn't know Rust to understand what is happening. I understand what you mean that Node can be difficult for scripting.

Reviewer: ```suggestion
JW: I would not say no to donated work done in Rust. OSS is an opportunity for contributors to learn and grow, and that includes us. I would like if there was 1 other person who is active, to whom we can ask [Rust] questions. And also we could schedule a code walkthrough so that we can see how easy it would be for someone who doesn't know Rust to understand what is happening. I understand what you mean that Node can be difficult for scripting.
```

CM: I avoid TypeScript for public things for similar reasons

RM: Ruby caused a lot of friction. TypeScript would cause a lot of friction.

IL: language doesn't matter too much. Whichever script we end up having, there needs to be onboarding. For example, existing bash script was hard to review.

JG: Mostly agree with Jen and Isaac. Want to be able to help review. I volunteer to help understand it! Interested to learn.

RM: I am happy to pair with people and set aside some time for this.

JW: Chris, does this address your concerns? Do you still have any objections.

CM: Yes this addresses my concerns, since this a very-internal tool, it seems ok that it is Rust. External tools however, I feel strongly about JS. 

JW: This summary below could maybe be a handbook thing.

Summarize: We have some flexibility with programming languages used for internal tools. Things that we want people besides ourselves to contribute to need to be JavaScript. Scripts in general should have some onboarding so that others can review.

> ### release delays google spreadsheet (CM, 5 mins)

Reviewer: ```suggestion
### Google spreadsheet for release delays (CM, 5 mins)
```

> I just want to go over the document that I shared with the team last week in a meeting and make sure that we all understand what it is telling us, and hopefully we will be able to figure out some strategies to improve the situation.

Reviewer: ```suggestion
I just want to go over the document that I shared with the team on June 26 and make sure that we all understand what it is telling us. Hopefully we will be able to figure out some strategies to improve the situation.
```

> Takeaway: it's important to look at the gap between libraries in addition to the overal number of days late. We'll have a look to see if the reminder bot helps.

Reviewer: ```suggestion
Takeaway: it's important to look at the gap between libraries in addition to the overall number of days late. We'll have a look to see if the reminder bot helps.
```

# July 7, 2020

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [ ] Chris Manson (CM)
- [ ] Isaac Lee (IL)
- [ ] Jared Galanis (JG)
- [ ] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)

> - [ ] Ricard Mendes (RM)

Reviewer: ```suggestion
- [ ] Ricardo Mendes (RM)
```

- [ ] Robert Wagner (RW)

### Guests
- [] Your name here (INITIALS)

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

### Review PR queue
See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

> <!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->

Author: ```suggestion
### release scripts and automation (RM, 15mins)
<!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->
```

# July 7, 2020

Facilitator & Note Taker: 

## Attendees

Add yourself to the list if you attend and check the box!

### Core Team
- [ ] Amy Lam (AL)
- [ ] Chris Manson (CM)
- [ ] Isaac Lee (IL)
- [ ] Jared Galanis (JG)
- [ ] Jen Weber (JW)
- [ ] Jessica Jordan (JJ)
- [ ] Kenneth Larsen (KL)
- [ ] Melanie Sumner (MS)
- [ ] Ricard Mendes (RM)
- [ ] Robert Wagner (RW)

### Guests
- [] Your name here (INITIALS)

## Weekly review

### Review action items (weekly)
See: [Action items board](https://github.com/orgs/ember-learn/projects/47)

### Review Initiatives board (every other week)
See: [Initiatives board](https://github.com/orgs/ember-learn/projects/19)

### Review PR queue
See: [tracking sheet](https://docs.google.com/spreadsheets/d/1sPyN9z9wZMpTNwqCfa6R9QSPZkIW4iQd-H4gZC7ILLk/edit#gid=2035777454)

## Topics

### release scripts and automation (RM, 15mins)

> <!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->

Author: ```suggestion
### tracked property caching
see: https://github.com/ember-learn/guides-source/pull/1474

<!-- If you would like to add a topic to the agenda please add a suggestion to the PR using the following format: -->
```
